Source/WebCore/dom/ScriptExecutionContext.h

```diff
@@ -286,6 +286,12 @@ class ScriptExecutionContext : public SecurityContext, public CanMakeWeakPtr<Scr
 
     ScriptExecutionContextIdentifier contextIdentifier() const;
 
+
+    // The following addMessage function is deprecated.
+    // Callers should try to create the ConsoleMessage themselves.
+    virtual void addMessage(MessageSource, MessageLevel, const String& message, const String& sourceURL, unsigned lineNumber, unsigned columnNumber, RefPtr<Inspector::ScriptCallStack>&&, JSC::JSGlobalObject* = nullptr, unsigned long requestIdentifier = 0) = 0;
+    virtual void logExceptionToConsole(const String& errorMessage, const String& sourceURL, int lineNumber, int columnNumber, RefPtr<Inspector::ScriptCallStack>&&) = 0;
+
```

Comment:
made these public for the sake of being able to delegate to them from the implementations in `ShadowRealmGlobalScope` but if we can provide an alternate implementation there, they can stay private